src/borg/cache.py

```diff
@@ -70,6 +70,10 @@ def __init__(self, repository, key, manifest, path=None, sync=True, do_files=Fal
         self.key = key
         self.manifest = manifest
         self.path = path or os.path.join(get_cache_dir(), repository.id_str)
+        if os.environ.get('BORG_UNIQ_HOSTNAME'):
```

Comment:
Frequently enough hostname is NOT something that is cloned with a VM, because it's obtaned via a reverse DNS lookup, for example.

e.g. I have a single subdir that holds an OS instance that I export with NFS, I then run tens of virtual machines out of it (nfsroot), every one of those virtual machines has a different hostname because of different DNS record.